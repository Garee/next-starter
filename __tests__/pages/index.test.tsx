import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Index from "../../pages/index";

describe("index page", () => {
  const allPosts = [
    {
      id: "1",
      date: "2021-01-01",
      title: "Post #1",
      content: "Sample content.",
      rating: 3,
    },
  ];

  const props = { greeting: "Hey ðŸ‘‹", description: "", allPosts };
  const index = <Index {...props} />;

  it("renders correctly", () => {
    expect.assertions(1);
    const tree = renderer.create(index).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with author header", () => {
    expect.assertions(1);
    render(index);
    const heading = screen.getByText("Gary Blackwood");
    expect(heading).toBeInTheDocument();
  });
});
