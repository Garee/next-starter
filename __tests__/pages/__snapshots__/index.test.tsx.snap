// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page renders correctly 1`] = `
<div
  className="container homeContainer"
>
  <header
    className="header"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQ0IiBoZWlnaHQ9IjE0NCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="Gary Blackwood"
        className="borderCircle"
        decoding="async"
        src="/_next/image?url=%2Fimages%2Favatar.jpg&w=384&q=75"
        srcSet="/_next/image?url=%2Fimages%2Favatar.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Favatar.jpg&w=384&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
    <h1
      className="heading2Xl"
    >
      Gary Blackwood
    </h1>
  </header>
  <main>
    <section
      className="headingMd"
    >
      <p>
        Hey ðŸ‘‹
      </p>
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <p>
        Check out this
         
        <a
          href="/projects"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          example page
        </a>
         
        and the
         
        <a
          href="/admin"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          CMS
        </a>
        .
      </p>
    </section>
    <section
      className="headingMd padding1px"
    >
      <h2
        className="headingLg"
      >
        Blog
      </h2>
      <ul
        className="list"
      >
        <li
          className="listItem"
        >
          <a
            href="/posts/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Post #1
          </a>
          <br />
          <small
            className="lightText"
          >
            <time
              dateTime="2021-01-01"
            >
              January 1, 2021
            </time>
          </small>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
